compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -506,6 +506,15 @@ static LogicalResult setWarpReductionConfig(func::FuncOp entryPoint,
                     {64, uint64_t(groupSize)}, {64, uint64_t(maxWorkgroupSize)})
                     .getZExtValue();
   }
+  // Current warp reduction pattern is a two step butterfly warp reduce.
+  // First, do warp reductions along multiple warps.
+  // Second, reduce results from multiple subgroups using single warp reduce.
+  // The final warp reduce requires numWarpsUsed > cudaWarpSize to work.
+  // TODO(raikonenfnu): Add flexible num of warp reduce to handle more configs.
+  const int64_t numWarpsUsed = groupSize / cudaWarpSize;
+  if (numWarpsUsed > cudaWarpSize) {
+    return failure();
+  }
```

Comment:
This is impossible on both cuda and rocm so I think we can just rely on the assert you have in the pass